Assets/HoloToolkit/UX/Scripts/BoundingBoxes/BoundingBox.cs

```diff
@@ -0,0 +1,498 @@
+            get
+            {
+                return targetBoundsWorldCenter;
+            }
+        }
+
+        /// <summary>
+        /// The world scale of the target object's bounds
+        /// Use this if you want to get the size of the target object
+        /// </summary>
+        public Vector3 TargetBoundsScale
+        {
+            get
+            {
+                return scaleTransform.localScale;
+            }
+        }
+
+        /// <summary>
+        /// The local scale of the target object's bounds
+        /// </summary>
+        public Vector3 TargetBoundsLocalScale
+        {
+            get
+            {
+                return targetBoundsLocalScale;
+            }
+        }
+
+        /// <summary>
+        /// The current flattened axis, if any
+        /// </summary>
+        public virtual FlattenModeEnum FlattenedAxis
+        {
+            get
+            {
+                return flattenedAxis;
+            }
+            protected set
+            {
+                if (flattenedAxis != value)
+                {
+                    flattenedAxis = value;
+                    if (OnFlattenedAxisChange != null) {
+                        OnFlattenedAxisChange();
+                    }
+                }
+            }
+        }
+
+        #endregion
+
+        #region protected
+
+        [Header("Objects")]
+        [Tooltip("The target object")]
+        [SerializeField]
+        protected GameObject target;
+
+        [Tooltip("The transform used to scale the bounding box (will be auto-generated)")]
+        [SerializeField]
+        protected Transform scaleTransform = null;
+
+        [Header("Flattening & Padding")]
+        [Tooltip("Flattening behavior setting.")]
+        [SerializeField]
+        protected FlattenModeEnum flattenPreference = FlattenModeEnum.FlattenAuto;
+
+        [Tooltip("The relative % size of an axis must meet before being auto-flattened")]
+        [SerializeField]
+        protected float flattenAxisThreshold = 0.025f;
+
+        [Tooltip("The relative % size of a flattened axis")]
+        [SerializeField]
+        protected float flattenedAxisThickness = 0.01f;
+
+        [Tooltip("How much to pad the scale of the box to fit around objects (as % of largest dimension)")]
+        [SerializeField]
+        protected float scalePadding = 0.05f;
+
+        [Tooltip("How much to pad the scale of the box on an axis that's flattened")]
+        [SerializeField]
+        protected float flattenedScalePadding = 0f;
+
+        [Header("Bounds Calculation")]
+        [Tooltip("Method used to calculate the bounds of the object.")]
+        [SerializeField]
+        protected BoundsCalculationMethodEnum boundsCalculationMethod = BoundsCalculationMethodEnum.MeshFilterBounds;
+
+        [Tooltip("Any renderers on this layer will be ignored when calculating object bounds")]
+        [SerializeField]
+        [Range(0, 31)]
+        protected int ignoreLayer = 2;//Ignore Raycast
```

Comment:
Should we be using the same layers as the InputManager?

Reply:
Not in this case. This is a layer that's used for choosing which objects to use when calculating bounds, not for input or raycasts.

Reply:
Then why the Ignore Raycast comment?

Reply:
That's the name of the layer. It's one of the default physics layers.

Reply:
Should this be a LayerMask so we can choose more than one?

Reply:
That's a good idea.